### PR TITLE
Add in-app messaging and native push notifications support

### PR DESCRIPTION
**Pull request by cgwyllie**
- Add support for in-app messaging based on updating base SDKs
- Handle all push notification features without requiring any 3rd-party push plugins

### Breaking Changes

- Initialisation is now required in native Android, native iOS, and JavaScript for full feature coverage
- `Kumulos.pushTrackOpen` has been removed (now handled by the SDK)
- Push registrations & notifications are now handled by the SDK so existing 3rd-party push plugins are no longer required

### Release Checklist

Prepare:

-   [x] Detail any breaking changes. Breaking changes require a new major version number
-   [ ] Check `pod lib lint` passes

Bump versions in:

-   [x] `package.json`
-   [x] `src/ios/KumulosReactNative.m`
-   [x] `src/android/.../KumulosReactNative.java`
-   [x] `README.md`

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
- [ ] Fill out release notes

### CODE REVIEW

File: src/ios/KumulosReactNative.m [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+static KSPushReceivedInForegroundHandlerBlock ksPushReceivedHandler;
+static KSPushNotification* _Nullable ksColdStartPush;
+
+static const NSString* KSReactNativeVersion = @"5.0.0";
+static const NSUInteger KSSdkTypeReactNative = 9;
+static const NSUInteger KSRuntimeTypeReactNative = 7;
+
+@implementation KumulosReactNative
+
+//- (dispatch_queue_t)methodQueue
```
Comment (azlindsay) [inline src/ios/KumulosReactNative.m]: Delete?

Comment (cgwyllie) [reply src/ios/KumulosReactNative.m]: Can do? Is part of the boilerplate RN plugin stuff so had just left it there forever

File: src/consts.js [hunk 2]
```diff
@@ -1,30 +1,8 @@
-
-const ReactPackage = require('react-native/package.json');
-const KumulosPackage = require('../package.json');
-
 export const BUILD_BASE_URL = 'https://api.kumulos.com';
 export const PUSH_BASE_URL = 'https://push.kumulos.com';
 
 export const CrashReportFormat = 'raven';
 
-const RuntimeType = {
```
Comment (azlindsay) [inline src/consts.js]: Are we at risk of the issue from objC SDK with these numbers being misread sometimes?

File: src/android/src/main/java/com/kumulos/reactnative/KumulosReactNative.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        KumulosInApp.updateConsentForUser(consented);
+    }
+
+    @ReactMethod
+    public void inAppGetInboxItems(Promise promise) {
+        SimpleDateFormat formatter;
+        formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", Locale.US);
+        formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        List<InAppInboxItem> inboxItems = KumulosInApp.getInboxItems(reactContext);
```
Comment (azlindsay) [inline src/android/src/main/java/com/kumulos/reactnative/KumulosReactNative.java]: Can the JS layer talk directly to the java InAppInboxItems? I thought i saw a DTO for push notifications elsewhere

File: src/consts.js (hunk 2 above)
Comment (cgwyllie) [reply src/consts.js]: Runtime info now read from consts in the ObjC/Java files

File: src/android/src/main/java/com/kumulos/reactnative/KumulosReactNative.java (hunk 3 above)
Comment (cgwyllie) [reply src/android/src/main/java/com/kumulos/reactnative/KumulosReactNative.java]: Everything has to be copied into a DTO as said, they don't share any address space so can't reference the same models

**Pull request merged**